test/e2e/serial/tests/operator_profile_update.go

```diff
@@ -0,0 +1,282 @@
+
+	Context("using the NUMA-aware scheduler without updated NRT data", func() {
+		// TODO The set of tests under this context are expecting that all worker nodes of the cluster
+		// be NROP-configured. this case is covered. However, if the cluster has other worker nodes (>2)
+		// and those nodes are not NROP-configured then the expected behavior of these tests will change
+		// and currently is missing and likely to be addressed in https://issues.redhat.com/browse/CNF-10552.
+		var testPod *corev1.Pod
+		nropObjInitial := &nropv1.NUMAResourcesOperator{}
+		nroKey := objects.NROObjectKey()
+
+		BeforeEach(func() {
+			By("disable RTE functionality hence publishing updated NRT data")
+			err := fxt.Client.Get(context.TODO(), nroKey, nropObjInitial)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q from the cluster", nroKey.String())
+
+			updatedNROPObj := nropObjInitial.DeepCopy()
+			infoRefreshPauseMode := nropv1.InfoRefreshPauseEnabled
+			//TODO adapt the existence of multiple node groups,https://issues.redhat.com/browse/CNF-10552
+			updatedNROPObj.Spec.NodeGroups[0].Config = &nropv1.NodeGroupConfig{
+				InfoRefreshPause: &infoRefreshPauseMode,
+			}
+
+			err = fxt.Client.Update(context.TODO(), updatedNROPObj)
+			Expect(err).ToNot(HaveOccurred())
+			klog.Info("wait long enough to verify NROP object is updated")
+			Eventually(func() nropv1.InfoRefreshPauseMode {
+				nropObjCurrent := &nropv1.NUMAResourcesOperator{}
+				err = fxt.Client.Get(context.TODO(), nroKey, nropObjCurrent)
+				Expect(err).ToNot(HaveOccurred(), "post NROP updated: cannot get %q from the cluster", nroKey.String())
+				//TODO adapt the existence of multiple node groups,https://issues.redhat.com/browse/CNF-10552
+				currentMode := nropObjCurrent.Status.MachineConfigPools[0].Config.InfoRefreshPause
+				if currentMode != &infoRefreshPauseMode {
+					klog.Warningf("nodegroup status is not completely updated yet. Expecting InfoRefreshPause to equal %q but found %q", &infoRefreshPauseMode, currentMode)
+				}
+				return *currentMode
+			}).WithTimeout(2*time.Minute).WithPolling(9*time.Second).Should(Equal(infoRefreshPauseMode), "failed to update the NROP object")
+
+			klog.Info("wait for the ds to get its args updated")
+			dsKey := wait.ObjectKey{
+				Namespace: nropObjInitial.Status.DaemonSets[0].Namespace,
+				Name:      nropObjInitial.Status.DaemonSets[0].Name,
+			}
+			Eventually(func() []string {
+				dsObj := appsv1.DaemonSet{}
+				err = fxt.Client.Get(context.TODO(), client.ObjectKey(dsKey), &dsObj)
+				Expect(err).ToNot(HaveOccurred())
+				cnt := k8swgobjupdate.FindContainerByName(dsObj.Spec.Template.Spec.Containers, manifests.ContainerNameRTE)
+				Expect(cnt).NotTo(BeNil(), "cannot find container data for %q", manifests.ContainerNameRTE)
+				return cnt.Args
+
+			}).WithTimeout(time.Minute).WithPolling(9*time.Second).Should(ContainElement(ContainSubstring("--no-publish")), "ds was not updated as expected: \"--no-pubish\" arg is missing.")
```

Comment:
this is misplaced. Another test (perhaps controller test) must test that if we set the pause field in the API object THEN the `--no-publish` command line argument pops in the DS command line args. We should not retest this here, but depend on the fact this specific bit works.

Reply:
I actualy recall now why I added this, it is because if the test proceed to the next step of checking if ds is ready, and finds I as so, it is likely that it still has the old config. so there were 2 ways: 1. wait for ds pods to get recreated=wait for ds to become not ready Or 2. wait for the new config to be reflected and only then wait for the ds to get ready. 
I chose to go with the second option because if the check failed this options gives us the bottom of the problem which is whether new config is reflected or not.